client/my-sites/posts/post-controls.jsx

```diff
@@ -1,226 +1,237 @@
-	Gridicon = require( 'components/gridicon'),
-	recordEvent = require( 'lib/analytics' ).ga.recordEvent;
-
-module.exports = React.createClass( {
-	displayName: 'PostControls',
-
-	mixins: [ PureRenderMixin ],
-
-	propTypes: {
-		post: React.PropTypes.object.isRequired,
-		editURL: React.PropTypes.string.isRequired,
-		onShowMore: React.PropTypes.func.isRequired,
-		onHideMore: React.PropTypes.func.isRequired,
-		onPublish: React.PropTypes.func,
-		onTrash: React.PropTypes.func,
-		onDelete: React.PropTypes.func,
-		onRestore: React.PropTypes.func,
-	},
-
-	view: function() {
-		recordEvent( 'Posts', 'Clicked View Post' );
-	},
-
-	preview: function() {
-		recordEvent( 'Posts', 'Clicked Preview Post' );
-	},
-
-	edit: function() {
-		recordEvent( 'Posts', 'Clicked Edit Post' );
-	},
-
-	viewStats: function() {
-		recordEvent( 'Posts', 'Clicked View Post Stats' );
-	},
-
-	buildControls: function( controlsArray ) {
-		var postId = this.props.post.ID,
-			controls;
-
-		controls = controlsArray.map( function( item, i ) {
-			var itemKey = 'controls-' + postId + '-' + i;
-
+import config from 'config';
+import utils from 'lib/posts/utils';
+import Gridicon from 'components/gridicon';
+import { ga } from 'lib/analytics';
+
+class PostControls extends PureComponent {
+
+	static propTypes = {
+		post: PropTypes.object.isRequired,
+		editURL: PropTypes.string.isRequired,
+		onShowMore: PropTypes.func.isRequired,
+		onHideMore: PropTypes.func.isRequired,
+		onPublish: PropTypes.func,
+		onTrash: PropTypes.func,
+		onDelete: PropTypes.func,
+		onRestore: PropTypes.func,
+		translate: PropTypes.func,
+	};
+
+	view() {
+		ga.recordEvent( 'Posts', 'Clicked View Post' );
+	}
+
+	preview() {
+		ga.recordEvent( 'Posts', 'Clicked Preview Post' );
+	}
+
+	edit() {
+		ga.recordEvent( 'Posts', 'Clicked Edit Post' );
+	}
+
+	copy() {
+		ga.recordEvent( 'Posts', 'Clicked Copy Post' );
+	}
+
+	viewStats() {
+		ga.recordEvent( 'Posts', 'Clicked View Post Stats' );
+	}
+
+	buildControls( controls ) {
+		return controls.map( ( control, i ) => {
 			return (
-				<li key={ itemKey }>
-					<a href={ item.href } className={ item.className } onClick={ item.onClick } target={ item.target ? item.target : null }>
-						<Gridicon icon={ item.icon } size={ 18 } />
-						<span>{ item.text }</span>
+				<li key={ `controls-${ this.props.post.ID }-${ i }` }>
+					<a
+						href={ control.href }
+						className={ `post-controls__${ control.className }` }
+						onClick={ control.onClick }
+						target={ control.target ? control.target : null }
+					>
+						<Gridicon icon={ control.icon } size={ 18 } />
+						<span>
+							{ control.text }
+						</span>
 					</a>
 				</li>
 			);
-		}, this );
-
-		return controls;
-	},
+		} );
+	}
 
-	render: function() {
-		var post = this.props.post,
-			availableControls = [],
-			extraControls = [],
-			parsed, previewURL, statsURL, postControlsClass, mainControls, moreControls, i, moreControlsSpliceIndex;
+	setControls() {
```

Comment:
Ouch, I totally missed that!
I'm gonna think about it (which likely means: "I'm going with your suggestions" 😆 ) while we wait for the wp-desktop confirmation about the class name.